dotcom-rendering/src/web/lib/useSWRPromise.ts

```diff
@@ -0,0 +1,32 @@
+import type { Fetcher } from 'swr';
+import useSWR from 'swr';
+
+/**
+ * We created this as a wrapper to get a promise from SWR.
+ * This was to prevent a complete refactor of the existing braze code,
+ * which expected a promise.
+ *
+ * @deprecated prefer refactoring the code to use SWR natively
+ */
+export const useSWRPromise = <Data extends unknown, Key extends string>(
```

Comment:
We only use this code in two places so can we just duplicate it and not have the abstraction? If this code is already deprecated and we want to discourage it's use then we would achieve that much better by simply not creating it?

Also, inlining things is going to make it easier to read the code without the need to jump around files, which is always nice

_And_, if we don't need to pass things around so much we will need fewer types, it's all good stuff.

Reply:
Yes, the ideal would be to not create it… however this would mean rewriting the slots’s braze logic completely, which we felt would be best achieved in a subsequent PR.

When pairing on it, using generics helped us confirm that the compiler was in agreement with what we were trying to achieve. We definitely did not want to inline this code because while we strongly believe it needs to go, having two copies of a code that does something we don’t want is worse than a single copy.

Reply:
If we know we don't want people to use this abstraction then not creating it is probably the best way to stop that happening. Creating an abstraction that we don't want used is metaphorically asking for trouble.

Alternatively, you could rename it to a name that is less inviting? `useSWRPromise_DEPRECATED`? Although I feel it is weird to create something which is immediately deprecated

Reply:
Would it be possible here to use a plain TS function which memoizes the returned `Promise`? I think we have that pattern elsewhere in DCR. (I guess you'd need to combine with `useEffect`/`useState` to ensure this happens at the right time. This also might be relevant to Max's comment below, if we want to ensure `buildBrazeMessages` is always called: 

> When it comes to making sure something is always called on a page, the current approach is having a script in the browser folder, like we do for bootCmp https://github.com/guardian/dotcom-rendering/pull/3661, initDiscussion https://github.com/guardian/dotcom-rendering/pull/3873 and [coreVitals](https://github.com/guardian/dotcom-rendering/blob/main/dotcom-rendering/src/web/browser/coreVitals/init.ts)

Reply:
I actually prefer this being broken out in and purposely deprecated, and I'd support oliver's idea of naming it `useSWRPromise_DEPRECATED`

Why?

copy-pasting code is not a rare thing in DCR, so many people work on this codebase it's commonplace to go look for patterns in other places and use them for your own component/function. This is fine, most of this code will be abstracted when and if it's needed, so it's no big deal to go looking for an example of what you're trying to do.

The advantage of abstracting this pattern is that we can mark it globally as something not to use, so if in a years time this pattern still exists for some reason, but we're all gone, if somebody were to copy and paste the way we were using SWR to generate a promise, there's a decent chance nobody would even realise it was a deprecated pattern.

Having a function, clearly named `useSWRPromise_DEPRECATED` would make it explicitly clear for any future developers that this is not a pattern we want to support. There's no chance it accidentally ends up in another place, at least not without the developers explicitly understanding why it was their only option. 

Therefor I feel like it's the _safer_ way to introduce this code, without risking it evolving or being assumed 'okay' in the future.